src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const states = [];
+
+  let currentState = { ...state };
+
+  for (let i = 0; i < actions.length; i++) {
+    const action = actions[i];
+    let nextState;
+
+    if (action.type === 'clear') {
+      nextState = {};
+    } else if (action.type === 'addProperties') {
+      // eslint-disable-next-line max-len
+      nextState = { ...currentState, ...action.extraData };
+    } else if (action.type === 'removeProperties') {
+      nextState = { ...currentState };
+
+      for (let j = 0; j < action.keysToRemove.length; j++) {
+        delete nextState[action.keysToRemove[j]];
+      }
+    }
```

Comment:
[CODE STYLE]: Use switch statement if you have a limited amount of conditions. This will improve the readability and organization of your code. Convert the if-else-if chain into a switch statement.